### PR TITLE
Remove all wcsaxes references.

### PR DESCRIPTION
**Pull request by Cadair**
This commit enforces an Astropy >= 1.3 dependency

### CODE REVIEW

**Comment by Cadair** [comment]
@pep8speaks quiet

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -285,15 +285,8 @@ def _as_mpl_axes(self):
...
-        try:
-            from wcsaxes import WCSAxes
-        except ImportError:
-            raise ImportError("Using WCS instances as Matplotlib projections "
-                              "requires the WCSAxes package to be installed. "
-                              "See http://wcsaxes.readthedocs.io for more "
-                              "details.")
-        else:
-            return WCSAxes, {'wcs': self.wcs}
+        from astropy.visualisation.wcsaxes import WCSAxes
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: This import doesn't need to be in here any more.

**Comment by wtbarnes** [comment]
@Cadair I think you misspelled "visualization". Tests seem to be failing because
```Python
ImportError: No module named visualiastion
```

**Comment by Cadair** [comment]
@ayshih I am having issues with the figure tests here. It seems that it's not generating me the figure test images for the failing tests (it's only generating me 4 pngs and 9 tests are failing).

**Comment by ayshih** [comment]
@Cadair Ha ha, that's an amusing bug!  Currently, each PNG is created with a temporary filename, and the proper test name is determined using a reverse lookup by the hash.  When there's a mismatched hash, the image isn't renamed/moved.  Another fix to add to #1961, I guess!

**Comment by ayshih** [comment]
@Cadair Your figure-hash commit (24948c9) needs to be reverted.  In fact, I've produced the very figures with those new hashes outside of this PR, and they have messed-up axes.

**Comment by ayshih** [comment]
To show you what I mean, here's the PNG for `test_draw_contours_aia` with the hash that starts `78c...`
sunpy map tests test_mapbase test_draw_contours_aia
compared to the correct PNG with the hash that starts with `c16...`
sunpy map tests test_mapbase test_draw_contours_aia

**Comment by Cadair** [comment]
Oh dear! We should merge your other PR so I can see the figures before I blindly merge the hash changes!

**Comment by ayshih** [comment]
To be able to merge #1961 satisfactorily before this PR, I need to be able to stay with Astropy 1.2.1 in the environment file, so I pushed a change to `sunpy/sunpy-figure-tests`.  Once this PR is merged, we can clean up the environments on that repo.

**Comment by Cadair** [comment]
+1

**Comment by Cadair** [comment]
This now includes the commits in #1969, the figure tests will fail until we update the envs in the figure test repo.

**Comment by Cadair** [comment]
@ayshih I added the env yml's in this PR now and the figure test builds pass.

**Comment by Cadair** [comment]
@sunpy/sunpy-developers review please!

File: doc/source/guide/data_types/maps.rst [hunk 2]
```diff
@@ -178,12 +178,13 @@ examples see :ref:`plotting`.
...
 .. note::
 
-   If the `wcsaxes <http://wcsaxes.readthedocs.io/en/latest/>`_ package is not
-   installed the `~sunpy.map.GenericMap.plot()` and `~sunpy.map.GenericMap.peek()`
-   methods assume that the data is not rotated,
-   i.e. the solar y axis is oriented with the columns of the array. If this condition
-   is not met, when the map is plotted a warning will be issued. You can create
-   an oriented map by using `~sunpy.map.GenericMap.rotate()` before you plot the Map.
+   If the `astropy.visualization.wcsaxes` package is not used (which is the
+   default) the `~sunpy.map.GenericMap.plot()` and
```
Comment (ayshih) [inline doc/source/guide/data_types/maps.rst]: Which way is the default: not using the package or using the package?  Suggest rephrase.

File: doc/source/guide/data_types/maps.rst [hunk 3]
```diff
@@ -178,12 +178,13 @@ examples see :ref:`plotting`.
...
+   `~sunpy.map.GenericMap.peek()` methods assume that the data is not rotated,
+   i.e. the solar y axis is oriented with the columns of the array. If this
+   condition is not met, when the map is plotted a warning will be issued. You
```
Comment (ayshih) [inline doc/source/guide/data_types/maps.rst]: This should probably instead say that a warning is issued if metadata indicates that the data needs to be rotated.  After all, this assumption may be invalid on some data sets, but if the metadata doesn't say so, we won't be able to issue a warning.

File: doc/source/guide/plotting.rst [hunk 4]
```diff
@@ -198,9 +199,10 @@ instances are created explicitly, and then used to modify the plot:
     plt.colorbar()
     plt.show()
 
-It is possible to create the same plot, explicitly not using `wcsaxes`, however,
-this will not have the features of `wcsaxes` which include correct
-representation of rotation and plotting in different coordinate systems.
+It is possible to create the same plot, explicitly not using
+`~astropy.visualizatiion.wcsaxes`, however, this will not have the features of
+`~astropy.visualizatiion.wcsaxes` which include correct representation of
```
Comment (ayshih) [inline doc/source/guide/plotting.rst]: Fix spelling of "visualization" on both lines

**Comment by Cadair** [comment]
travis fails are expected

**Comment by Cadair** [comment]
Currently debugging the python 3.5 figure test fails

**Comment by Cadair** [comment]
Travis is good to go again.

**Comment by Cadair** [comment]
@sunpy/sunpy-maintainers This PR is just screaming *merge meeee!* ;)

File: doc/source/requirements.txt [hunk 5]
```diff
@@ -11,4 +11,3 @@ pytest
 mock
 sqlalchemy
 sphinx_rtd_theme
-wcsaxes>=0.4
```
Comment (Punyaslok) [inline doc/source/requirements.txt]: Wrong astropy version in Line 9 above.

Comment (Punyaslok) [inline doc/source/requirements.txt]: Wrong astropy version in Line 9.

Comment (Cadair) [inline doc/source/requirements.txt]: Fixed.

**Comment by Cadair** [comment]
I now feel bad, should have let @Punyaslok press the button :p

**Comment by Punyaslok** [comment]
LOL

**Pull request merged**